File: Profile Picture Downloader For Instagrame/Instagram-profile-Downloader/app/src/main/java/com/example/profilepicturedownloaderforinstagrame/activities/MainActivity.java
package com.example.profilepicturedownloaderforinstagrame.activities;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.AsyncTask;
import android.os.Bundle;
import android.util.Log;
import android.view.View;
import android.widget.Button;
import android.widget.ImageView;
import android.widget.TextView;
import android.widget.Toast;

import com.example.profilepicturedownloaderforinstagrame.AppPrefrences;
import com.example.profilepicturedownloaderforinstagrame.Dialog.AuthenticationDialog;
import com.example.profilepicturedownloaderforinstagrame.R;
import com.example.profilepicturedownloaderforinstagrame.interfaces.AuthenticationListener;
import com.squareup.picasso.Picasso;

import org.apache.http.HttpEntity;
import org.apache.http.HttpResponse;
import org.apache.http.client.HttpClient;
import org.apache.http.client.methods.HttpGet;
import org.apache.http.impl.client.DefaultHttpClient;
import org.apache.http.util.EntityUtils;
import org.json.JSONException;
import org.json.JSONObject;

import java.io.IOException;

public class MainActivity extends AppCompatActivity implements AuthenticationListener {
    String token = null;
    Button btn_login_instagram_account;
    AuthenticationDialog authenticationDialog = null;
    AppPrefrences appPrefrences = null;
    View info = null;
    Button btn_searchingActivity;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        btn_login_instagram_account = findViewById(R.id.login_instagram_account);
        btn_searchingActivity=findViewById(R.id.searchingActivity);
        btn_searchingActivity.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                Intent i=new Intent(MainActivity.this,DownloadProfileImageActivity.class);
                startActivity(i);
            }
        });
        appPrefrences = new AppPrefrences(this);
        token = appPrefrences.getString(AppPrefrences.TOKEN);
        if (token != null) {
            getUserInfoByAccessToken(token);
        }
    }

    public void login() {
        btn_login_instagram_account.setText("LOGOUT");
        info.setVisibility(View.VISIBLE);
        ImageView pic = findViewById(R.id.pic);
        Picasso.with(this).load(appPrefrences.getString(AppPrefrences.PROFILE_PIC)).into(pic);
        TextView id = findViewById(R.id.id);
        id.setText(appPrefrences.getString(AppPrefrences.USER_ID));
        TextView name = findViewById(R.id.name);
        name.setText(appPrefrences.getString(AppPrefrences.USER_NAME));
    }

    public void logout() {
        btn_login_instagram_account.setText("LOGIN INSTAGRAM");
        token = null;
        info.setVisibility(View.VISIBLE);
        appPrefrences.clear();
    }

    @Override
    public void onTockenRecieved(String auth_tocken) {
        if (auth_tocken == null)
            return;
        appPrefrences.putString(AppPrefrences.TOKEN, auth_tocken);
        token = auth_tocken;
        getUserInfoByAccessToken(token);
    }

    public void onClick(View view) {
        if (token != null) {
            logout();
        } else {
          /*  authenticationDialog = new AuthenticationDialog(MainActivity.this, this);
            authenticationDialog.setCancelable(true);
            authenticationDialog.show();*/
          Intent i=new Intent(MainActivity.this,AuthenticationActivity.class);
          startActivity(i);
            /*Toast.makeText(this, "Show", Toast.LENGTH_SHORT).show();*/
        }
    }
    private void getUserInfoByAccessToken(String token) {
        new RequestInstagramApi().execute();
    }

    public class RequestInstagramApi extends AsyncTask<Void, String, String> {

        @Override
        protected void onPreExecute() {
            super.onPreExecute();
        }

        @Override
        protected String doInBackground(Void... voids) {
            HttpClient httpClient = new DefaultHttpClient();
            HttpGet httpGet = new HttpGet(getResources().getString(R.string.get_user_info_url) + token);
            try {
                HttpResponse response = httpClient.execute(httpGet);
                HttpEntity httpEntity = response.getEntity();
                return EntityUtils.toString(httpEntity);
            } catch (IOException e) {
                e.printStackTrace();
            }
            return null;
        }

        @Override
        protected void onPostExecute(String response) {
            super.onPostExecute(response);
            if (response != null) {
                try {
                    JSONObject jsonObject = new JSONObject(response);
                    Log.e("response", jsonObject.toString());
                    JSONObject jsonData = jsonObject.getJSONObject("data");
                    if (jsonData.has("id")) {
                        appPrefrences.putString(AppPrefrences.USER_ID, jsonData.getString("id"));
                        appPrefrences.putString(AppPrefrences.USER_NAME, jsonData.getString("username"));
                        appPrefrences.putString(AppPrefrences.PROFILE_PIC, jsonData.getString("profile_picture"));
                        login();
                    }
                } catch (JSONException e) {
                    e.printStackTrace();
                }
            }
        }
    }
}

File: CalculatorApp/app/src/main/java/com/example/calculatorapp/MainActivity.kt
package com.example.calculatorapp

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.Button
import android.widget.EditText
import android.widget.TextView

class MainActivity : AppCompatActivity() {

    var no1:Float=0.0F
    var no2:Float=0.0F

    var number1:EditText?=null
    var number2:EditText?=null
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        number1=findViewById<EditText>(R.id.number1)
        number2=findViewById<EditText>(R.id.number2)

        val total=findViewById<TextView>(R.id.total)


        findViewById<Button>(R.id.add).setOnClickListener{
            getNumbers()
            total.text="Total = ${no1+no2} "
        }
        findViewById<Button>(R.id.subtract).setOnClickListener{
            getNumbers()
            total.text="Total = ${no1-no2} "
        }
        findViewById<Button>(R.id.mul).setOnClickListener{
            getNumbers()
            total.text="Total = ${no1*no2} "
        }
        findViewById<Button>(R.id.div).setOnClickListener{
            getNumbers()
            total.text="Total = ${no1/no2} "
        }

        findViewById<Button>(R.id.clear).setOnClickListener{
            no1=0.0F
            no2=0.0F
            number1?.setText("")
            number2?.setText("")
            number1?.hint = "e.g 123"
            number2?.hint = "e.g 432"
            total.text="Total = 0.0"
        }
        findViewById<Button>(R.id.exit).setOnClickListener{
            finish()
        }

    }

    fun getNumbers(){
        no1=(number1?.text.toString()).toFloat()
        no2=(number2?.text.toString()).toFloat()
    }
}